snippets/language-crystal.cson

```diff
@@ -0,0 +1,406 @@
+  'map_with_index { |e, i| .. }':
+    'prefix': 'map_with_index'
+    'body': 'map_with_index { |${1:e}, ${2:i}| $0 }'
+  'max':
+    'prefix': 'max'
+    'body': 'max'
+  'max_by { |e| .. }':
+    'prefix': 'max_by'
+    'body': 'max_by { |e| $0 }'
+  'min':
+    'prefix': 'min'
+    'body': 'min'
+  'min_by { |e| .. }':
+    'prefix': 'min_by'
+    'body': 'min_by { |e| $0 }'
+  'minmax':
+    'prefix': 'minmax'
+    'body': 'minmax'
+  'none?':
+    'prefix': 'none?'
+    'body': 'none?'
+  'module .. end':
+    'prefix': 'mod'
+    'body': 'module ${1:ModuleName}\n\t$0\nend'
+  'raise exception':
+    'prefix': 'raise exception'
+    'body': 'raise ${1:exception}'
+  'raise message':
+    'prefix': 'raise message'
+    'body': 'raise "${1:message}"'
+  'rand':
+    'prefix': 'rand'
+    'body': 'rand'
+  'reduce { |memo, e| .. }':
+    'prefix': 'reduce'
+    'body': 'reduce(${1:default}) { |${2:memo}, ${3:e}| $0${2:memo} }'
+  'require ".."':
+    'prefix': 'require'
+    'body': 'require "$0"'
+  'reverse':
+    'prefix': 'reverse'
+    'body': 'reverse'
+  'scan(/../) { |match| .. }':
+    'prefix': 'scan'
+    'body': 'scan(/${1:pattern}/) { |${2:match}| $0 }'
+  'select { |e| .. }':
+    'prefix': 'select'
+    'body': 'select { |${1:e}| $0 }'
+  'sort':
+    'prefix': 'sort'
+    'body': 'sort'
+  'ternary_if':
+    'prefix': 'ternary_if'
+    'body': '${1:case} ? ${2:result1} : ${3:result3}\n'
+  'unless … end':
+    'prefix': 'unless'
+    'body': 'unless (${1:condition})\n\t$0\nend'
+  'when …':
+    'prefix': 'when'
+    'body': 'when ${1:condition}\n\t$0'
+  'yield':
+    'prefix': 'yield'
+    'body': ' :yield ${0:arguments}'
+# Classes
+  'Array':
+    'prefix': 'Array'
+    'body': 'Array'
+  'Bool':
+    'prefix': 'Bool'
+    'body': 'Bool'
+  'Char':
+    'prefix': 'Char'
+    'body': 'Char'
+  'Class':
+    'prefix': 'Class'
+    'body': 'Class'
+  'Enum':
+    'prefix': 'Enum'
+    'body': 'Enum'
+  'Enumerable':
+    'prefix': 'Enumerable'
+    'body': 'Enumerable'
+  'Float':
+    'prefix': 'Float'
+    'body': 'Float'
+  'Float32':
+    'prefix': 'Float32'
+    'body': 'Float32'
+  'Float64':
+    'prefix': 'Float64'
+    'body': 'Float64'
+  'Int':
+    'prefix': 'Int'
+    'body': 'Int'
+  'Int32':
+    'prefix': 'Int32'
+    'body': 'Int32'
+  'Int64':
+    'prefix': 'Int64'
+    'body': 'Int64'
+  'JSON':
+    'prefix': 'JSON'
+    'body': 'JSON'
+  'Nil':
+    'prefix': 'Nil'
+    'body': 'Nil'
+  'Object':
+    'prefix': 'Object'
+    'body': 'Object'
+  'Proc':
+    'prefix': 'Proc'
+    'body': 'Proc'
+  'Range':
+    'prefix': 'Range'
+    'body': 'Range'
+  'Regex':
+    'prefix': 'Regex'
+    'body': 'Regex'
+  'Set':
+    'prefix': 'Set'
+    'body': 'Set'
+  'String':
+    'prefix': 'String'
+    'body': 'String'
+  'Struct':
+    'prefix': 'Struct'
+    'body': 'Struct'
+  'Symbol':
+    'prefix': 'Symbol'
+    'body': 'Symbol'
+  'Time':
+    'prefix': 'Time'
+    'body': 'Time'
+  'Tuple':
```

Comment:
I'm not sure snippets are intended for class autocompletion, but for now it will work well.